### PR TITLE
Provide a documentation with YARD

### PR DESCRIPTION
**Pull request by vbyno**
Document DSL and NotNil modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,98 @@
...
+```ruby
+# Gemfile
+gem 'custom_active_record_observer'
 ```
-CustomActiveRecordObserver.observe :EngagedUser, :CompanyOwner, handler: CustomerEventPublisher.new do
-  on_create do |customer|
-    Events::CustomerSignedUp.new(data: {customer_id: customer.id}
+and run `bundle install`
+
+Then add path to that file to `InfoHub` paths (in `config/initializers/events_observation.rb`)
```
Comment (lesykgg) [inline README.md]: Infohub?

File: README.md [hunk 2]
```diff
@@ -1,25 +1,98 @@
...
-  on_destroy do |customer|
-    Events::CustomerSignedOut.new(data: {…})
+We can track by real classes (which is not always possible, since that code is better to be placed in `config/initializers`. But passing strings or symbols is also allowed:
+```ruby
+  CustomActiveRecordObserver.observe User, :EngagedUser, 'CompanyOwner' { ... }
+```
+Here we track three classes.
+
+### Usage
+How do we use `custom_active_record_observer` in the wild? For instance, in this example we publish `UserRegisteredEvent` on user creation. (using `rails_event_store` gem). This is pretty useful trade-off if there are no operations or aggregates introduced in the system achitecture.
```
Comment (lesykgg) [inline README.md]: typo

**Review by lesykgg — COMMENTED** [review]
:+1:

**Review by lesykgg — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,25 +1,98 @@
...
+Then add path to that file to `InfoHub` paths (in `config/initializers/events_observation.rb`)
+```ruby
+CustomActiveRecordObserver.observe 'User' do
+  on_create do |user|
+    # do something
   end
+end
+```
+
+In this expample we track model `User`. If a new user is created, AR `:after_commit` callback is launched and the block in `on_create` executes.
```
Comment (lesykgg) [inline README.md:30]: typo

**Review by dkh — APPROVED** [review]

**Review by myacheg — APPROVED** [review]

**Pull request merged**